src/main/java/org/broadinstitute/hellbender/tools/longreads/graph/AlignedBaseGraphCollection.java

```diff
@@ -0,0 +1,583 @@
+                else if ( graph.outDegreeOf(vertex) == 0 ) {
+                    endVertices.add((AlignedBaseVertex) vertex);
+                }
+            }
+        }
+
+        // Now we can check the starts / ends to see if they're next to eachother.
+        // If so, we can add a link between them:
+        for ( final AlignedBaseVertex endVertex : endVertices ) {
+            for ( final AlignedBaseVertex startVertex : startVertices ) {
+                if ( endVertex.isAdjacentTo(startVertex) ) {
+                    graph.addEdge(endVertex, startVertex);
+                }
+            }
+        }
+    }
+
+    /**
+     * Add in the given aligned sequence information to this {@link AlignedBaseGraphCollection}.
+     * @param read An aligned {@link GATKRead} object to add to this {@link AlignedBaseGraphCollection}.
+     */
+    public void addSequence(final GATKRead read) {
+
+        final byte[]             bases    = read.getBasesNoCopy();
+        final List<CigarElement> cigar    = read.getCigarElements();
+        final String             contig   = read.getContig();
+        final int                startPos = read.getStart();
+        final String             readName = read.getName();
+
+        addSequence(bases, cigar, contig, startPos, readName);
+    }
+
+    /**
+     * Add in the given aligned sequence information to this {@link AlignedBaseGraphCollection}.
+     * This information is assumed to be pulled directly from a {@link GATKRead} object.  This has implications for
+     * certain parameters (the {@code startPos} in particular).
+     *
+     * Data can only be added to this graph before it is collapsed.  After that, it is an error to add anything else.
+     *
+     * @param bases Bases to be added in the order in which they appear in the read.
+     * @param cigar {@link List<CigarElement>} containing alignment information for each base in {@code bases}.
+     * @param contig The contig to which the given base sequence is aligned.
+     * @param startPos The start position of the first aligned base in the given base sequence.
+     * @param readName The name of the parent read from which this aligned sequence information is derived.
+     */
+    public void addSequence(final byte[] bases,
+                            final List<CigarElement> cigar,
+                            final String contig,
+                            final int startPos,
+                            final String readName) {
+
+        // Make sure we're still OK to add data:
+        if ( isGraphCollapsed ) {
+            throw new GATKException("Attempted to addSequence to " + this.getClass().getName() + " after collapsing it.  This is a logic error.");
+        }
+
+        // Create a queue of nodes that represent the linear graph of all nodes in the given data:
+        final Queue<AlignedBaseVertex> nodeQueue = createAlignedNodes(bases, cigar, contig, startPos, readName);
+
+        if ( nodeQueue.peek() != null ) {
```

Comment:
I think you want to ask nodeQueue.isEmpty() or some such. You don't apparently have any way of putting null nodes in here but thats another factor to consider.